jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/BuildReferenceDateUtilTest.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.kit.common.util;
+
+import org.junit.Test;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.time.Instant;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class BuildReferenceDateUtilTest {
+  @Test
+  public void testStoreReferenceDateInPluginContext() {
+    // Given
+    Date currentDate = new Date();
+    Map<String, Object> pluginContext = new HashMap<>();
+
+    // When
+    BuildReferenceDateUtil.storeReferenceDateInPluginContext(currentDate, pluginContext, "buildTimestampContextKey");
+
+    // Then
+    assertEquals(1, pluginContext.size());
+    assertTrue(pluginContext.get("buildTimestampContextKey") instanceof Date);
+  }
+
+  @Test
+  public void testGetBuildReferenceDateWhenFileDoesntExist() {
+    assertNotNull(BuildReferenceDateUtil.getBuildReferenceDate("target", "docker"));
+  }
+
+  @Test
+  public void testGetBuildReferenceDate() throws IOException {
+    // Given
+    File buildDirectory = Files.createTempDirectory("test-project").toFile();
```

Comment:
Please use this instead

  @Rule
  public TemporaryFolder temporaryFolder = new TemporaryFolder();